libr/bin/format/elf/elf.c

```diff
@@ -1726,7 +1726,7 @@ ut64 Elf_(r_bin_elf_get_main_offset)(ELFOBJ *bin) {
 	/* linux64 pie main -- probably buggy in some cases */
 	if (buf[29] == 0x48 && buf[30] == 0x8d) { // lea rdi, qword [rip-0x21c4]
 		ut8 *p = buf + 32;
-		st32 maindelta = p[0] | p[1]<<8 | p[2]<<16 | p[3]<<24;
+		st32 maindelta = (ut32)p[0] | (ut32)p[1]<<8 | (ut32)p[2]<<16 | (ut32)p[3]<<24;
```

Comment:
use r_read_le32() instead of manual work

`st32 maindelta = (st32) r_read_le32 (buf + 32);`